### PR TITLE
SIMSBIOHUB-18: Fix application redirection links

### PR DESCRIPTION
**Pull request by curtisupshall**
# Overview

Enhances URL redirection in the SIMS frontend.

## Links to Jira tickets

 - https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-18

## Description of relevant changes

- Fixes the application URL provided in Access Request email links
- Adds a login endpoint to hook into Keycloak sign-in
- Adds the `useRedirect` hook

### CODE REVIEW

File: app/src/components/layout/Header.tsx [hunk 1]
```diff
@@ -224,44 +230,48 @@ const Header: React.FC = () => {
...
+          </Container>
+        </Box>
       </AppBar>
 
       <Dialog open={open}>
         <DialogTitle>Contact Support</DialogTitle>
         <DialogContent>
           <Typography variant="body1" component="div" color="textSecondary">
-            For technical support or questions about this application, please email&nbsp;
+            For technical support or questions about this application, please email &zwnj;
```
Comment (jeznorth) [inline app/src/components/layout/Header.tsx]: Why the change here? Just curious.

File: app/src/pages/landing/LandingPage.tsx [hunk 2]
```diff
@@ -1,15 +1,178 @@
...
-  });
+  const hasAdministrativeRole = keycloakWrapper?.hasSystemRole([
+    SYSTEM_ROLE.DATA_ADMINISTRATOR,
+    SYSTEM_ROLE.SYSTEM_ADMIN
+  ]);
 
-  return null;
+  return (
+    <BaseLayout className={classes.baseLayoutContainer}>
+      <Container maxWidth="md">
```
Comment (jeznorth) [inline app/src/pages/landing/LandingPage.tsx]: We're going to have to come up with a better strategy here - this layout seems odd on larger viewports.

Comment (jeznorth) [reply app/src/pages/landing/LandingPage.tsx]: Encapsulate the content with a negative top margin to align visual center here, not at the containing element.

File: .config/config.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
   "staticUrls": {
     "dev": "dev-biohubbc.apps.silver.devops.gov.bc.ca",
     "test": "test-biohubbc.apps.silver.devops.gov.bc.ca",
-    "prod": "biohubbc.apps.silver.devops.gov.bc.ca"
+    "prod": "sims.nrs.gov.bc.ca"
```
Comment (NickPhura) [inline .config/config.json:23]: We'll need to specifically test this change in OpenShift. 

It might be fine, but it might also cause trouble if the pods need the internal openshift route to communicate properly. If that ends up being the case, we could add a 4th field here called something like "prod-vanity-url", and then it can pass the sims URL to the appHost env var, and the rest of them can use the original internal url.

File: app/src/components/security/RouteGuards.tsx [hunk 4]
```diff
@@ -119,26 +119,14 @@ export const UnAuthenticatedRouteGuard = (props: RouteProps) => {
...
-      const redirectUrlParams = qs.stringify(urlParams, { filter: (prefix) => prefix !== 'authLogin' });
-      const redirectUri = `${window.location.origin}${location.pathname}?${redirectUrlParams}`;
-
-      // Trigger login
-      keycloakWrapper?.keycloak?.login({ redirectUri: redirectUri });
-    }
-
-    return <Redirect to="/" />;
+    // Trigger login, then redirect to the desired route
+    return <Redirect to={`/login?next=${encodeURIComponent(location.pathname)}`} />;;
```
Comment (NickPhura) [inline app/src/components/security/RouteGuards.tsx]: nit: the keycloak and siteminder redirects use `redirect_uri` and `returl` respectively. Maybe we can change `next` to something more similar: `redirect` or `redirect_uri` seem like good options, just so its more clear what this field is for.

File: app/src/hooks/useRedirect.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { useCallback, useMemo } from 'react';
+import { buildUrl } from 'utils/Utils';
+import { useQuery } from './useQuery';
+
+interface Redirect {
+  redirectUri: string | undefined;
+  redirect: () => void;
+}
+
+export default function useRedirect(fallback?: string): Redirect {
```
Comment (NickPhura) [inline app/src/hooks/useRedirect.tsx:10]: Nice 👍

File: app/src/components/security/RouteGuards.tsx (hunk 4 above)
Comment (curtisupshall) [reply app/src/components/security/RouteGuards.tsx]: Changed to `redirect`.

**Review by AlfredRosenthal — APPROVED** [review]
🦞

File: app/src/components/layout/Header.tsx (hunk 1 above)
Comment (curtisupshall) [reply app/src/components/layout/Header.tsx]: Just to avoid the nagging `{' '}` literal

**Review by anissa-agahchen — APPROVED** [review]
